Sources/Sentry/SentryProfiler.mm

```diff
@@ -100,15 +100,20 @@
         threadMetadata[threadID] = metadata;
     }
     if (!backtrace.threadMetadata.name.empty() && metadata[@"name"] == nil) {
-        metadata[@"name"] = [NSString stringWithUTF8String:backtrace.threadMetadata.name.c_str()];
+        const auto name = [NSString stringWithUTF8String:backtrace.threadMetadata.name.c_str()];
+        if (name != nil) {
+            metadata[@"name"] = name;
```

Comment:
we don't need this for a subscript assignment. `metadata[key] = nil` will not crash like it will when you put it in a literal, it effectively just removes the element if it exists or does nothing otherwise

Reply:
ah yes, thank you for pointing this out

Reply:
Also [`stringWithUTF8String`](https://developer.apple.com/documentation/foundation/nsstring/1497379-stringwithutf8string) doesn't return `nil`. It raises an exception, though, when you pass `NULL` to it. You already check the `name` with `empty()`. So I don't think this will change any behavior.